src/Extractor/Rst.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rusty\Extractor;
+
+use Gregwar\RST\ErrorManager;
+use Gregwar\RST\HTML\Nodes\CodeNode;
+use Gregwar\RST\Parser;
+use League\CommonMark\Block\Element;
+use Rusty\CodeSample;
+use Rusty\PragmaParser;
+
+class Rst implements SampleExtractor
+{
+    protected $pragmaParser;
+
+    public function __construct()
+    {
+        $this->pragmaParser = new PragmaParser();
+    }
+
+    public static function supportedExtensions(): array
+    {
+        return ['rst'];
+    }
+
+    public function extractSamples(\SplFileInfo $file): \Traversable
+    {
+        $parser = new Parser();
+
+        $parser->getEnvironment()->setErrorManager(new class() extends ErrorManager {
+            public function error($message)
+            {
+
+            }
+        });
+
+        $fileContent = file_get_contents($file->getRealPath());
+
+        /** @var \Gregwar\RST\HTML\Document $document */
+        $document = @$parser->parse($fileContent);
```

Comment:
When reading the rst file, the parser tries to read the include statements and then reading those files without testing if the file exists. So without this, and defining a current directory to the parser. A lot of warnings could occur and they will be useless because rusty will already find all the files in the folder.